File: 01-super-basic-questions/04-factorial/main.py
num = int(input("Enter a number: "))

factResult = 1

def factorial(n):
    if n == 0:
        return 1
    return n * factorial(n - 1)


for i in range(num, 1, -1):
    factResult *= i
# print("Factorial of ", num, "is ", factResult)
print("Factorial of {} is {}".format(num, factResult))
print("Factorial of {} is {} using recurssion".format(num, factResult))

File: 01-super-basic-questions/04-factorial/main.go
package main

import "fmt"

func main() {
	var (
		num int
	)

	fmt.Printf("Enter the number: ")
	//fmt.Scanln(&input)
	fmt.Scanf("%d", &num)

	factResult := 1
	for i := num; i >= 2; i-- {
		factResult *= i
	}

	fmt.Printf("Factorial of %d is %d\n", num, factResult)
	fmt.Printf("Factorial of %d is %d using Recurssion\n", num, factorial(num))

}

func factorial(num int) int {
	if num == 0 {
		return 1
	}

	return num * factorial(num-1)
}
